lib/Macros/Sources/ObservationMacros/ObservableMacro.swift

```diff
@@ -58,11 +58,15 @@ public struct ObservableMacro {
   static func withMutationFunction(_ observableType: TokenSyntax) -> DeclSyntax {
     return 
       """
-      internal nonisolated func withMutation<Member, MutationResult>(
-        keyPath: KeyPath<\(observableType), Member>,
-        _ mutation: () throws -> MutationResult
-      ) rethrows -> MutationResult {
-        try \(raw: registrarVariableName).withMutation(of: self, keyPath: keyPath, mutation)
+      @_disfavoredOverload
+      internal nonisolated func withMutation<Member>(
+          keyPath: KeyPath<ObsClass , Member>,
+          storageKeyPath: ReferenceWritableKeyPath<ObsClass , Member>,
+          newValue: Member
+      ) {
+        try \(raw: registrarVariableName).withMutation(of: self, keyPath: keyPath) {
+          self[keyPath: storageKeyPath] = newValue
```

Comment:
will this pose a problem with access performance? or behavior? (the latter I don't think it should be too much of an issue)

Reply:
I didn't change the `access `, so it shouldn't have changed.
For behavior, my change can be equated to this: put the creation of the closure from the setter into the withMutation.